compiler/src/utils/wasm_utils.re

```diff
@@ -259,6 +259,20 @@ let serialize_abi_version = ({major, minor, patch}) => {
   bytes;
 };
 
+let utf8_encode = (ints) => {
+  let buf = Buffer.create(14);
```

Comment:
It's always 14?

Reply:
It's just an initial guess for how many characters you think there will be in the buffer, and then it resizes accordingly. Sorta like hash tables.

Reply:
Does the buffer ever shrink though? I thought it only grew

Reply:
That's true. It only grows unless you call `Buffer.reset` on it, and then it shrinks back to the initial size you told it to be (and throws away everything in your buffer). That size parameter though is for the amount of space it allocates to store buffer contents, and that's separate from how much stuff is actually in the buffer. I imagine it's got some sort of internal state that it keeps track of that.

Here's a quick test from the REPL:
```ocaml
# let buf = Buffer.create 14;;
val buf : Buffer.t = <abstr>
# Buffer.add_string buf "hello";;
- : unit = ()
# Buffer.contents buf;;
- : string = "hello"
```